src/x509.c

```diff
@@ -8635,8 +8635,16 @@ WOLF_STACK_OF(WOLFSSL_X509)* wolfSSL_X509_chain_up_ref(
                 return WOLFSSL_FAILURE;
             }
 
+            cert->decodedCert = (DecodedCert*)XMALLOC(sizeof(DecodedCert), cert->heap,
+                                                               DYNAMIC_TYPE_TMP_BUFFER);
+            if (cert->decodedCert == NULL)
+                return MEMORY_E;
```

Comment:
This will leak `cert` and `rng`. You need to add `XFREE(cert, NULL, DYNAMIC_TYPE_CERT);` and `wc_FreeRng(&rng)` or change this logic.

Reply:
Ahh yes. Added free's for both